src/settings/PatronNotices/PatronNoticeDetail.test.js

```diff
@@ -0,0 +1,312 @@
+import React from 'react';
+import {
+  render,
+  screen,
+  within,
+  fireEvent,
+} from '@testing-library/react';
+
+import '../../../test/jest/__mock__';
+
+import {
+  Accordion,
+  AccordionSet,
+  Button,
+  KeyValue,
+} from '@folio/stripes/components';
+import {
+  PreviewModal,
+  tokensReducer,
+} from '@folio/stripes-template-editor';
+
+import PatronNoticeDetail from './PatronNoticeDetail';
+
+const mockParseWithInstructionsReturnValue = 'parsedValue';
+const mockTokensReducerReturnValue = 'tokensReducerValue';
+
+jest.mock('@folio/stripes-template-editor', () => ({
+  PreviewModal: jest.fn(({ onClose }) => (
+    // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions
+    <div data-testid="previewModal" onClick={onClose} />
+  )),
+  tokensReducer: jest.fn(() => mockTokensReducerReturnValue),
+}));
+jest.mock('html-to-react', () => ({
+  __esModule: true,
+  default: {
+    ProcessNodeDefinitions: jest.fn(() => ({
+      processDefaultNode: 'defaultNode',
+    })),
+  },
+  Parser: () => ({
+    parseWithInstructions: jest.fn(() => mockParseWithInstructionsReturnValue),
+  }),
+}));
+Button.mockImplementation(({ children, onClick }) => (
```

Comment:
I believe that global `Button` mock will be enough here. It also render `children` and have `onClick` property.